### PR TITLE
add store/add fetch data logic

### PR DESCRIPTION
**Pull request by ElhamMoharrami**

### CODE REVIEW

File: note.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//provider set on index.js
```
Comment (ghamilouie) [inline note.txt:1]: Can you remove this file and keep it local?

File: src/components/ArtistList/ArtistList.js [hunk 2]
```diff
@@ -1,70 +1,56 @@
...
-  //change items per page on select
-const itemsPerPageHandler=(selectedOption)=>{
-  setItemsPerPage(selectedOption[0].value)
-}
-
-
-  //for the first time that the page loads so we have something to show
-  //calculates the total number of pages we need for pagination
+  const dispatch=useDispatch()
+
```
Comment (ghamilouie) [inline src/components/ArtistList/ArtistList.js:15]: Do we need these blank lines here? (applies to all other files)

File: src/components/ArtistList/ArtistList.js [hunk 3]
```diff
@@ -1,70 +1,56 @@
...
+   
     const getArtistRequest = async () => {
-      const url = `http://localhost:8080/api/crews?page=1&size=${itemsPerPage}`;
-      
-      const response = await fetch(url);
-      const data = await response.json();
-      const total = data.page.totalElements;
-      setPageCount(Math.floor(total / itemsPerPage));
-      setArtists(data.content);
+      const url = `http://localhost:8080/api/crews?page=${currentPage}&size=${itemsPerPage}`;
```
Comment (ghamilouie) [inline src/components/ArtistList/ArtistList.js:27]: Isn't it better to extract this constant configs to a separate config file?

File: src/components/ArtistList/ArtistList.js [hunk 4]
```diff
@@ -1,70 +1,56 @@
...
-    window.scrollTo(0, 0);
+  const handlePageClick = async (event) => {
+    dispatch(comboActions.changeCurrentPage(event.selected + 1))
   };
 
-  //invoke when dropdown opens
+  const itemsPerPageHandler=(selectedOption)=>{
+    const selectedOpt=selectedOption[0].value;
+      dispatch(comboActions.changeItemsPerPage(selectedOpt))
+
```
Comment (ghamilouie) [inline src/components/ArtistList/ArtistList.js:42]: Maybe another unnecessary blank line :)

File: src/components/movieList/MovieList.js [hunk 5]
```diff
@@ -1,69 +1,55 @@
...
-  }, [itemsPerPage]);
-
-  //runs when the page changes
-  const fetchMovieRequest = async (currentPage) => {
-    const url = `http://localhost:8080/api/movies?page=${currentPage}&size=${itemsPerPage}`;
-    const response = await fetch(url);
-    const data = await response.json();
-    return data.content;
+  const getMovieRequest = async () => {
+    const url = `http://localhost:8080/api/movies?page=${currentPage-1}&size=${itemsPerPage}`;
```
Comment (ghamilouie) [inline src/components/movieList/MovieList.js:29]: Isn't it better to extract this constant configs to a separate config file?

File: src/components/movieList/MovieList.js [hunk 6]
```diff
@@ -1,69 +1,55 @@
-import React, { Fragment, useEffect, useState } from "react";
+import React, { Fragment, useEffect} from "react";
```
Comment (ghamilouie) [inline src/components/movieList/MovieList.js:1]: Directory name "movieList" doesn't match the naming convention.

File: src/store/data-slice.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import { createSlice} from '@reduxjs/toolkit';
```
Comment (ghamilouie) [inline src/store/data-slice.js:1]: Code format doesn't look pretty and neat(ex. indentations don't match) you can use your idea reformat

File: src/store/data-slice.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  
+
+
+export const fetchData=(url)=>{
+
+    return async (dispatch)=>{
+
+     const  getDataRequest=async()=>{
+      const response = await fetch(url);
+      const data = await response.json();
```
Comment (ghamilouie) [inline src/store/data-slice.js:37]: Do we need `data` variable?

File: src/components/movieList/MovieList.js (hunk 5 above)
Comment (ghamilouie) [inline src/components/movieList/MovieList.js:29]: Can we replace ` with ' ?

File: src/components/ArtistList/ArtistList.js (hunk 3 above)
Comment (ghamilouie) [inline src/components/ArtistList/ArtistList.js:27]: Can we replace ` with ' ?

File: note.txt (hunk 1 above)
Comment (ElhamMoharrami) [reply note.txt:1]: note.txt was my own reminder sheet that I forgot to delete.file deleted.

File: src/components/ArtistList/ArtistList.js (hunk 2 above)
Comment (ElhamMoharrami) [reply src/components/ArtistList/ArtistList.js:15]: code formated with prettier. blank lines removed.

File: src/components/ArtistList/ArtistList.js (hunk 3 above)
Comment (ElhamMoharrami) [reply src/components/ArtistList/ArtistList.js:27]: it is not possible in this case.

File: src/components/movieList/MovieList.js (hunk 6 above)
Comment (ElhamMoharrami) [reply src/components/movieList/MovieList.js:1]: A component in React should always use a pascal case. I changed movieList to MovieList in my last commit.

File: src/store/data-slice.js (hunk 7 above)
Comment (ElhamMoharrami) [reply src/store/data-slice.js:1]: installed prettier.issue fixed.

File: src/store/data-slice.js (hunk 8 above)
Comment (ElhamMoharrami) [reply src/store/data-slice.js:37]: this is our data-slice. we get the data we want from api. also using data for this specific part is common. somehow a convention.

File: src/components/movieList/MovieList.js (hunk 5 above)
Comment (ElhamMoharrami) [reply src/components/movieList/MovieList.js:29]: no we cant.

File: src/components/ArtistList/ArtistList.js (hunk 3 above)
Comment (ElhamMoharrami) [reply src/components/ArtistList/ArtistList.js:27]: it is react syntax.

**Pull request merged**